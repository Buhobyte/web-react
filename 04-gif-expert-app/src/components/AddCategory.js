import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const    AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 1) {
            setCategories(categories => [inputValue, ...categories])
            setInputValue('');
        }
    }

    return (
        // <> no es necesario el fragment ya que form sirve de fragmentador o agrupador
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            {/* <h2>{inputValue}</h2> */}
            <input type='text' value={inputValue} onChange={handleInputChange}></input>
        </form>
        // </>
    )


}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
