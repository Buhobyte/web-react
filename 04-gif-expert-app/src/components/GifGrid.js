import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {


    const [count, setCount] = useState(0);

    useEffect(() => {
        dataGifAPI();
    }, [])

    const dataGifAPI = async () => {
        const api_key = 'IeSlcE9JlBLhzpNlK7PPFQHfkvcg0mjY';
        const resp = await fetch('https://api.giphy.com/v1/gifs/search?q=Rick&limit=10&api_key=' + api_key);
        const { data } = await resp.json()

        const dataCatagories = data.map((item) => {
            return {
                id: item.id,
                title: item.title,
                url: item.images?.downsized_medium.url,

            }
        })
        console.log(dataCatagories);
    }

    return (
        <div>
            <h2>{category}</h2>
            <h3>{count}</h3>
            <button onClick={() => { setCount(c => c + 1) }}></button>

        </div>
    )
}

GifGrid.propTypes = {
    categories: PropTypes.array
}

export default GifGrid
