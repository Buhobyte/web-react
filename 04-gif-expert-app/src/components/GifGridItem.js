import React from 'react'

import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className='item-card animate__fadeIn'>
            <img alt={title} src={url}></img>
            <p>{title}</p>
        </div>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}