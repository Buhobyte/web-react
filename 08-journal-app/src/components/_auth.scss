.auth_main {
    display: flex;
    background-color: $primary;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;
    margin: 0px;
    height: 100vh;
    width: 100vw;
}

.auth_box-container {
    background-color: white;
    box-shadow: 0px 3px $dark-grey;
    border-radius: 2px;
    padding: 20px;
}