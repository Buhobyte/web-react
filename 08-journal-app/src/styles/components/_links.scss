.link {
    text-decoration: none;
    columns: $dark-grey;

    &:hover {
        text-decoration: underline;
    }
}