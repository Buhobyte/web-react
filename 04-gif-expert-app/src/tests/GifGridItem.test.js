import React from 'react';

import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';


describe('Probar componentes', () => {
    const title = "titulo prueba"
    const url = "https://localhost.com"
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe mostrar <GifGridItem/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    })

    test('debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    })

    test('debe tener la imagen igual a la url y el title', () => {
        const img = wrapper.find('img');
        expect(img.prop('alt')).toBe(title)
        expect(img.prop('src')).toBe(url)
    })

    test('debe tener animate flhas_in', () => {
        const div = wrapper.find('div');
        const className = div.prop('className')

        expect(className.includes('animate__fadeIn')).toBe(true);
    })


})

