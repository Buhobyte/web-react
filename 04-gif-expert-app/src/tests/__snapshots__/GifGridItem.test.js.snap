// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar componentes debe mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="item-card animate__fadeIn"
>
  <img
    alt="titulo prueba"
    src="https://localhost.com"
  />
  <p>
    titulo prueba
  </p>
</div>
`;
